### PR TITLE
[xcvrd] add unit test infrastructure and unit tests for xcvrd

### PR DESCRIPTION
**Pull request by vdahiya12**
Summary:
This PR provides the necessary infrastructure to add pytest support and integration in sonic-xcvrd submodule.
This PR also adds unit tests for xcvrd daemon.
### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Approach
Added changes in the setup.py and setup.cfg and initialized the tests directory

	new file:   setup.cfg
	new file:   pytest.ini
	modified:   setup.py
	new file:   tests/__init__.py
	new file:   tests/mock_platform.py
	new file:   tests/mock_swsscommon.py
	new file:   tests/test_xcvrd.py
	new file:   tests/t0-sample-port-config.ini
  

#### What is the motivation for this PR?
To add the pytest unittest support in sonic-platform-daemon, sonix-xcvrd daemon as well as add some unit tests

#### How did you do it?
Added the changes in sonic-platform-daemons module

#### How did you verify/test it?
run the unit tests by running it as part xcvrd build 
make target/python-wheels/sonic_xcvrd-1.0-py2-none-any.whl

Signed-off-by: vaibhav-dahiya <vdahiya@microsoft.com>

### CODE REVIEW

File: sonic-xcvrd/setup.py [hunk 1]
```diff
@@ -21,10 +21,14 @@
         'enum34; python_version < "3.4"',
         'sonic-py-common',
     ],
-    setup_requires=[
+    setup_requires = [
```
Comment (jleveque) [inline sonic-xcvrd/setup.py:24]: Remove spaces around `=`, as this is technically a named parameter to the `setup()` function.

File: sonic-xcvrd/setup.py [hunk 2]
```diff
@@ -21,10 +21,14 @@
...
         'wheel'
     ],
-    classifiers=[
+    tests_require = [
```
Comment (jleveque) [inline sonic-xcvrd/setup.py:27]: Remove spaces around `=`, as this is technically a named parameter to the `setup()` function.

File: sonic-xcvrd/setup.py [hunk 3]
```diff
@@ -21,10 +21,14 @@
...
+        'pytest',
+        'pytest-cov',
+    ],
+    classifiers = [
```
Comment (jleveque) [inline sonic-xcvrd/setup.py:31]: Remove spaces around `=`, as this is technically a named parameter to the `setup()` function.

File: sonic-xcvrd/tests/test_xcvrd.py [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+"""
+os.environ["XCVRD_UNIT_TESTING"] = "1"
+load_source('y_cable_helper', scripts_path + '/xcvrd_utilities/y_cable_helper.py')
+from y_cable_helper import *
+from xcvrd.xcvrd import *
+
+PY3x = sys.version_info >= (3, 0)
+PYvX_DIR = "py3" if PY3x else "py2"
+
+PYTHON_INTERPRETTER = "python3" if PY3x else "python2"
```
Comment (jleveque) [inline sonic-xcvrd/tests/test_xcvrd.py:44]: Typo: `PYTHON_INTERPRETER`. Also, is this used?

**Review by jleveque — CHANGES_REQUESTED** [review]

File: sonic-xcvrd/setup.py (hunk 1 above)
Comment (vdahiya12) [reply sonic-xcvrd/setup.py]: fixed

File: sonic-xcvrd/setup.py (hunk 2 above)
Comment (vdahiya12) [reply sonic-xcvrd/setup.py]: fixed

File: sonic-xcvrd/setup.py (hunk 3 above)
Comment (vdahiya12) [reply sonic-xcvrd/setup.py]: fixed

File: sonic-xcvrd/tests/test_xcvrd.py (hunk 4 above)
Comment (vdahiya12) [reply sonic-xcvrd/tests/test_xcvrd.py]: fixed, removed

**Review by jleveque — APPROVED** [review]

**Pull request merged**